client/app/pages/queries/components/QueryVisualizationTabs.jsx

```diff
@@ -29,12 +61,28 @@ export default function QueryVisualizationTabs({
     );
   }
 
-  const orderedVisualizations = orderBy(visualizations, ["id"]);
+  const orderedVisualizations = useMemo(() => orderBy(visualizations, ["id"]), [visualizations]);
+  const isFirstVisualization = useCallback(visId => visId === orderedVisualizations[0].id, [orderedVisualizations]);
 
   return (
-    <Tabs {...tabsProps} animated={false} tabBarGutter={0} onChange={activeKey => onChangeTab(+activeKey)}>
+    <Tabs
+      {...tabsProps}
+      className="query-visualization-tabs"
+      animated={false}
+      tabBarGutter={0}
+      onChange={activeKey => onChangeTab(+activeKey)}
+      destroyInactiveTabPane
+    >
       {orderedVisualizations.map(visualization => (
-        <TabPane key={`${visualization.id}`} tab={visualization.name}>
+        <TabPane
+          key={`${visualization.id}`}
+          tab={(canDeleteVisualizations && !isFirstVisualization(visualization.id)) ? (
+            <TabWithDeleteButton
+              visualizationName={visualization.name}
```

Comment:
Let's move this condition to `TabWithDeleteButton` component: add prop `canDelete: PropTypes.bool`, and use it as:
```javascript
tab={(
  <TabWithDeleteButton
    visualizationName={visualization.name}
    canDelete={canDeleteVisualizations && !isFirstVisualization(visualization.id)}
    onDelete={() => onDeleteVisualization(visualization)}
  />
)}
```